### ISSUE TITLE
Full example in keras

### ISSUE THREAD

**Issue by dfalbel** [vanHavel/bp-mll-tensorflow#1]
Do you have a model definition that works in keras?

I'm having problemas because loss is becoming NaN , after 10 steps. I don't know if it's the gradients or anything like this.

**Comment by vanHavel** [comment]
I added a full example in Keras, using the yeast dataset for multilabel learning from [[1]](http://www.ecmlpkdd2008.org/sites/ecmlpkdd2008.org/files/pdf/workshops/pl/11.pdf).
Make sure that the matrices y_true and y_pred have the right format. Let's say that you have n datapoints and k different labels. Then y_pred must be an (n x k)-matrix with values between 0 and 1, and y_true must be an (n x k)-matrix which only takes on the values 0 and 1.

[[1]](http://www.ecmlpkdd2008.org/sites/ecmlpkdd2008.org/files/pdf/workshops/pl/11.pdf) Park, Sang-Hyeun, and Johannes Fürnkranz. "Multi-label classification with label constraints." Proceedings of the Proceedings of the ECML/PKDD-08 Workshop on Preference Learning (PL-08). 2008.

**Comment by dfalbel** [comment]
Thanks very much!! I'll test it tomorow, but it seems great!

**Comment by RichardSieg** [comment]
I also only get NaNs, although my labels are exactly like in your example, e.g., 

``` 
y_train[0]

array([ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,
        0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,
        0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  1.,  0.,
        0.,  0.,  0.,  1.,  0.], dtype=float32)
```

**Comment by vanHavel** [comment]
Can you post a minimal example where the loss returns NaNs?

**Comment by RichardSieg** [comment]
My mistake - I had zero vectors. Sorry! Thanks for the tool!
